WeakAuras/GenericTrigger.lua

```diff
@@ -3887,6 +3904,40 @@ function GenericTrigger.GetTriggerDescription(data, triggernum, namestable)
   end
 end
 
+do
+  -- Based on Code by DejaCharacterStats. Ugly code to figure out the
+  local class = select(2, UnitClass("player"))
+  if class == "DEMONHUNTER"
+      or class == "HUNTER" or class == "SHAMAN"
+      or class == "MAGE" or class == "PRIEST" or class == "WARLOCK"
+      or class == "DEATHKNIGHT" or class == "PALADIN" or class == "WARRIOR"
+    then
+    function WeakAuras.CalculatedGcdDuration()
+      local haste = GetHaste()
+      return max(0.75, 1.5 * 100 / (100+haste))
+    end
+  elseif class == "DRUID" then
+    function WeakAuras.CalculatedGcdDuration()
+      local id = GetShapeshiftFormID()
+      local haste = GetHaste()
+      return id == 1 and 1 or max(0.75, 1.5 * 100 / (100+haste))
+    end
+  elseif class == "MONK" then
+    function WeakAuras.CalculatedGcdDuration()
+      local spec = GetSpecialization() -- TODO check spec index
+      local primaryStat = select(6, GetSpecializationInfo(spec))
```

Comment:
GetSpecializationInfo is nil on classic (nor does monks, so this comment is trash :D)